### ISSUE TITLE
occur context canceled if use --watch

### ISSUE THREAD

**Issue by hatappi** [go-task/task#313]
- Task version: 2.8.0
- OS: macos 10.15.3

When I launch task with `--watch` option, occur `context canceled` occurs.

I prepared a project to reproduce.
https://github.com/hatappi/go-http-server

first step, launch http server

```sh
$ git clone https://github.com/hatappi/go-http-server
$ cd go-http-server
$ make install-task

# run http server using task
$ make run-task
./bin/task -w run
task: Started watching for tasks: run
go build -v -o ./dist/app main.go
command-line-arguments
./dist/app
2020/04/21 00:19:57 start http server
```

second step, change `main.go`.
The following will be output error

```sh
task: Failed to run task "run": context canceled
go build -v -o ./dist/app main.go
command-line-arguments
./dist/app
2020/04/21 00:23:15 start http server
```

Is this the behavior  expecting?

**Comment by andreynering** [comment]
Hi @hatappi,

Yeah, I think we can probably just suppress context errors from being printed in the terminal.

**Comment by hatappi** [comment]
HI @andreynering 
Thank you for response.
I tried to resolve it and created PR.
https://github.com/go-task/task/pull/317

Could you review it if you have a time?

**Comment by andreynering** [comment]
Fixed on #317.

**Comment by hatappi** [comment]
Thank you for merge my PR and add new release tag.
